### DISCUSSION TITLE
LazyVim local development

### DISCUSSION THREAD

**Discussion by dpetka2001** [LazyVim/LazyVim#1129] [Q&A] [answered]
I would like to start getting used to configure LazyVim as a local plugin. I forked the LazyVim repo and cloned the forked repo in my laptop under `~/projects/plugins/`. Then I installed a new starter template to get a clean slate for developing (which I can load seperately with NVIM_APPNAME). I then configured in `/lua/config/lazy.lua` the dev path 
```lua
dev = {
  path = "~/projects/plugins/",
}
```
and added a spec for LazyVim in `/lua/plugins/LazyVim.lua` with the following
```lua
return {
  {
    "LazyVim/LazyVim",
    dev = true,
    import = "lazyvim/plugins",
  },
}
```
while disabling the one in `/lua/config/lazy.lua` by commenting out the line `{ "LazyVim/LazyVim", dev = true, import = "lazyvim.plugins" }` . But when `lazy.nvim` tries to install LazyVim it tries to install it under `~/.local/share/LazyVimDev/lazy/LazyVim`. If I comment out the import line then `Lazy` shows that LazyVim is in dir `~/projects/plugins/LazyVim`. So how would I go about importing the lazyvim plugins from the local forked repo? Should I set `dev = true` for every plugin that Lazyvim pulls from the import command?

**Comment by folke** [comment, marked as answer]
I personally just set `dev = {"LazyVim"}` in my `lazy.lua`.

Then clone the LazyVim repo under `~/projects/LazyVim`

No other changes should be needed.

**Reply by dpetka2001** [reply]
Thank you very much @folke. I thought `dev` needed to be a `boolean` value and then set the `config.dev.path` for Lazy to pick up changes. Can you also set `dev` to whichever plugin you want to try locally?

**Reply by dpetka2001** [reply]
Oh it is under `patterns`. I missed that. Thank you very much again for your help.

**Reply by dpetka2001** [reply]
One more question for whenever you have time to answer. Let's say that you work on a branch for a fix/feature. How do you make Neovim aware of that branch and the changes it contains (so that you can test things out) in contrast to the main branch where the cloned repo resides? Or does Neovim automatically pick up the branch you are currently on and uses that as the local dev for LazyVim? Or do you just do the testing after you merge your feature branch into main and do in the main the testing?

**Reply by amaanq** [reply]
pretty sure you can just set the `branch` key in the spec

**Reply by dpetka2001** [reply]
OK the local dev branch always gets picked up by Lazy depending on which branch you already are. Good to know and tnx again.

**Reply by dpetka2001** [reply]
Yes @amaanq you can do that as well. I just experimented with it now. Thank you for your input.

**Reply by folke** [reply]
@dpetka2001 for dev plugins (provided with `dev=true` or with dev patterns), those directories are not managed by lazy, so you can just checkout a branch there and it will be used as is.
